### PR TITLE
Update answers.txt

### PR DESCRIPTION
**Pull request by ireishp**
Hey @yetanotheranand . 
Please review my work.

### CODE REVIEW

File: answers.txt [hunk 1]
```diff
@@ -1,4 +1,22 @@
 What is a gitignore file and what should it include for a Java project? 
+A gitignore file is a file which tells git which files/folders can be ignored in a project. 
+For a java project, the gitignore file can include a list of class files, package files (like .jar, .war, etc), and crash logs. 
+
+
 What does fast-forward in git mean?
+Git fast forward is when we try to merge a commit that can be reached by following a previous commit history. Hence, git simply moves the pointer forward as there is nothing new to merge together.
```
Comment (yetanotheranand) [inline answers.txt:7]: Important thing over here is `nothing new divergent work to merge together`

File: answers.txt [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
 What does fast-forward in git mean?
+Git fast forward is when we try to merge a commit that can be reached by following a previous commit history. Hence, git simply moves the pointer forward as there is nothing new to merge together. 
+
+
 Why should you have different branches?
+We have different branches so that when working in a team, each developer can work on his/her code independently, without affecting other team members' code. It also helps us to maintain different versions of the code.
+
+
 In a project how many remote Git repositories do you usually have?
+We usually have ONE remote repository.
```
Comment (yetanotheranand) [inline answers.txt:15]: It depends from project to project. 
If the project is small, and worked upon by the same team. It is good to have one repo.
On the other hand, if the project is large with various sub-modules which are not tightly coupled, and is worked upon by several teams with different release cycle. It is better to have multiple repo.

Interesting Read: https://git-scm.com/book/en/v2/Git-Tools-Submodules

**Review by yetanotheranand — CHANGES_REQUESTED** [review]
Please make the changes for Q2.

Also, please reply to Q4's comment with your reasoning or add more details to the answer.

**Comment by ireishp** [comment]
Updated the answers as suggested. Pl review and close this.

**Review by yetanotheranand — APPROVED** [review]
LGTM

**Pull request merged**